FrontEnd/FrontEnd.Web/Services/RecipeService.cs

```diff
@@ -22,11 +23,27 @@ public RecipeService(HttpClient httpClient, IConfiguration configuration)
         _httpClient.BaseAddress = new Uri(configuration.GetValue<string>("ApiSettings:RecipesApiBaseUrl")!);
     }
     
-    public async Task<IEnumerable<Recipe>> GetAllAsync()
+    public async Task<IEnumerable<Recipe>> GetAllAsync(string? searchTerm = null, IEnumerable<Tag>? tags = null, (int skip, int top)? pagination = null)
     {
-        return await _httpClient.GetFromJsonAsync<IEnumerable<Recipe>>("recipes") ?? throw new InvalidOperationException();
+        var requestUri = (searchTerm, tags, pagination) switch
```

Comment:
Instead of going through each possible permutation of variables, why not simply add a check for each one and either add a substring containing the value or a blank string (i.e. `requestUri += searchTerm ?? string.Empty` or something similar)?

Reply:
I'll just use a `UriBuilder` here :D I was expanding this bit by bit, but it definitely makes more sense now